src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> previous;
+        private Node<T> next;
+
+        public Node(Node<T> previous, T item, Node<T> next) {
+            this.previous = previous;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        addToTail(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            addToTail(value);
+        } else {
+            addToHead(value, findNode(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T oldValue = findNode(index).item;
```

Comment:
let's save the result of `findNode(index)` to the variable and use this result
don't call method `findNode(index)` twice

Reply:
Fixed